ai/src/ai/common/example.py

```diff
@@ -11,14 +11,36 @@
 import shutil
 from typing import Generic, Literal, TypeVar
 
-from pydantic import BaseModel
+from pydantic import BaseModel, field_validator
 
 
 class ExampleInput(BaseModel):
   prompt: str
   input_code: str | None = None
   line_number_target: int | None = None
 
+  @field_validator("prompt")
+  @classmethod
+  def is_required(cls, v):
+    if v == "":
+      raise ValueError("Field is required.")
+    return v
+
+  @field_validator("line_number_target", mode="before")
+  @classmethod
+  # Intentionally leave out annotations since the input value may be ambiguous.
+  def convert_empty_string_to_none(cls, v):
+    if isinstance(v, str) and v == "":
+      return None
+    return v
+
+  @field_validator("line_number_target", mode="after")
+  @classmethod
+  def line_number_greater_than_0(cls, v: int | None) -> int | None:
```

Comment:
you could also validate that input_code is defined and line_number doesn't go out of bounds on the input code

Reply:
Good idea. I tried to implement it but ran into a slight issue since I think there are times where the Example model gets initialized without input code. The input code is then later updated.

Ended up just checking out of bounds for exceeding line length only if the input code data has already been loaded in the file. Which I think handles the form validation case.